src/inmanta/module.py

```diff
@@ -1131,21 +1130,41 @@ def load_plugins(self) -> None:
 
         try:
             mod_name = self._meta["name"]
-            imp.load_package(const.PLUGINS_PACKAGE + "." + mod_name, plugin_dir)
+            for py_file in glob.glob(os.path.join(plugin_dir, "**", "*.py"), recursive=True):
+                fq_mod_name = self._get_mod_name_for_py_file(py_file, plugin_dir, mod_name)
+                # Already loaded
+                if fq_mod_name in sys.modules.keys():
+                    continue
+
+                LOGGER.debug("Loading module %s", fq_mod_name)
+                if py_file.endswith("__init__.py"):
+                    imp.load_package(fq_mod_name, os.path.dirname(py_file))
```

Comment:
imp package is deprecated, in favor of importlib.

Now would perhaps be a good time to switch (as I assume you have become somewhat of an expert in importlib)